### PR TITLE
Fix typo in -query-frontend.align-querier-with-step CLI flag

### PR DESCRIPTION
**Pull request by hi-rustin**
#### What this PR does
Fix typo in -query-frontend.align-querier-with-step CLI flag. Still keep the old one for backward compatibility.
#### Which issue(s) this PR fixes or relates to

Ref https://github.com/grafana/mimir/issues/2829

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by hi-rustin** [comment]
The conflict has been resolved.

File: pkg/frontend/querymiddleware/roundtrip.go [hunk 1]
```diff
@@ -49,7 +49,9 @@ type Config struct {
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&cfg.MaxRetries, "query-frontend.max-retries-per-request", 5, "Maximum number of retries for a single request; beyond this, the downstream error is returned.")
 	f.DurationVar(&cfg.SplitQueriesByInterval, "query-frontend.split-queries-by-interval", 24*time.Hour, "Split range queries by an interval and execute in parallel. You should use a multiple of 24 hours to optimize querying blocks. 0 to disable it.")
-	f.BoolVar(&cfg.AlignQueriesWithStep, "query-frontend.align-querier-with-step", false, "Mutate incoming queries to align their start and end with their step.")
+	f.BoolVar(&cfg.AlignQueriesWithStep, "query-frontend.align-queries-with-step", false, "Mutate incoming queries to align their start and end with their step.")
+	// TODO: Remove it after 2 releases.
```
Comment (pracucci) [inline pkg/frontend/querymiddleware/roundtrip.go:53]: This change will be included in Mimir 2.4.0, so we can remove in 2.6.0.

```suggestion
	// TODO: Remove it in Mimir 2.6.0.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@
 * [ENHANCEMENT] Distributor: Add single forwarding remote-write endpoint for a tenant (`forwarding_endpoint`), instead of using per-rule endpoints. This takes precendence over per-rule endpoints. #2801
 * [ENHANCEMENT] Added `err-mimir-distributor-max-write-message-size` to the errors catalog. #2470
 * [ENHANCEMENT] Add sanity check at startup to ensure the configured filesystem directories don't overlap for different components. #2828
+* [ENHANCEMENT] Query-frontend: CLI flag `-query-frontend.align-querier-with-step` renamed to `query-frontend.align-queries-with-step`. #2840
```
Comment (pracucci) [inline CHANGELOG.md:21]:
```suggestion
* [CHANGE] Query-frontend: CLI flag `-query-frontend.align-querier-with-step` has been deprecated. Please use `-query-frontend.align-queries-with-step` instead. #2840
```

**Review by pracucci — APPROVED** [review]
I manually tried it and it works great! Thanks! ❤️

**Comment by hi-rustin** [comment]
Thanks for your review!  💚 💙 💜 💛 ❤️

**Comment by pracucci** [comment]
> Thanks for your review! 💚 💙 💜 💛 ❤️

All the opposite... thanks to **YOU** for all the great work you're doing. I'm impressed! Keep pushing 😉

**Pull request merged**